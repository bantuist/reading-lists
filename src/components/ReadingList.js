import React, { Component } from 'react';
import './../App.css';

const fakeBooks = [
  {volumeInfo: {title: 'The Catcher in the Rye', authors: "J. D. Salinger", imageLinks: {thumbnail: 'http://books.google.com/books/content?id=PCDengEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api'}}}
];
class ReadingList extends Component {
  render() {
    const books = fakeBooks.map(book => {
      return (
        <li>
          <img 
            src={book.volumeInfo.imageLinks.thumbnail}
            alt="thumbnail"
          />
          {book.volumeInfo.title}
          {book.volumeInfo.authors}
          Reading List
        </li>
      );
    });
    return (
      <div className="col">
        <ul>
        {books}
        </ul>
      </div>
    );
  }
}

export default ReadingList;
