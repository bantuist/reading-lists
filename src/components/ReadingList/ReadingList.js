import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function handleKeyDown(onUpdateList, event, id) {
  // TODO: Add escape edit 
  const enter = 13;
  if (event.which === enter && event.target.value !== '') {
    onUpdateList(event, id);
  }
}
function ReadingList({ list, notifications, onEditList, onRemoveList, onUpdateList }) {
  let id = 'none';
  let name = 'none';
  let bookCount = 0;

  if (list) {
    id = list.id;
    name = list.name;
    bookCount = list.bookCount;
  }
  let listItemEdit = 'list-item-edit';
  let removeClassName = 'remove';

  // Toggle list editing and notifications
  if (notifications && notifications.editing === id) {
    listItemEdit += ' editing';
    removeClassName += ' show';
  }

  return (
    <div className="list-item-container" key={id}>
      <li className="list-item">
        <Link 
          to={{ 
          pathname: '/current-list', 
          state: { currentList: list }
          }}
          className='list-view'
        >
          <h2 className="list-name">{name} ({bookCount})</h2>
        </Link>
          <button className="update" onClick={() => onEditList(id, name)} />
          <button className={removeClassName} onClick={() => onRemoveList(id)} />
        <input
          autoFocus
          placeholder="Enter a new name"
          className={listItemEdit}
          defaultValue={name}
          data-id={id}
          onKeyDown={(event) => handleKeyDown(onUpdateList, event, id)}
          onBlur={(event) => onUpdateList(event, id)}
        />
      </li>
    </div>
  );
}

ReadingList.propTypes = {
  list: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    createdAt: PropTypes.number.isRequired,
    bookCount: PropTypes.number.isRequired,
    doneCount: PropTypes.number.isRequired,
    books: PropTypes.object,
  }),
  editing: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  notification: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  // Require?
  onUpdateList: PropTypes.func,
  onEditList: PropTypes.func,
  onRemoveList: PropTypes.func,
};

export default ReadingList;