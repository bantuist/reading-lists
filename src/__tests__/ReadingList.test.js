import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import ReadingList from '../components/ReadingList/ReadingList';
import renderer from 'react-test-renderer';

function log(msg) {
  require('fs').appendFileSync('./jest.log.js', msg + '\n', {'encoding': 'utf8'});
}

it('should render ReadingList', () => {
  log(Link);
  const tree = renderer.create(<ReadingList />).toJSON();
  expect(tree).toMatchSnapshot();
});

