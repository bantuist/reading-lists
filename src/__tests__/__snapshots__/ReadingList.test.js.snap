// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReadingList 1`] = `
<div
  className="list-item-container"
>
  <li
    className="list-item"
  >
    <a
      className="list-view"
      href="/current-list"
      onClick={[Function]}
    >
      <h2
        className="list-name"
      >
        none
         (
        0
        )
      </h2>
    </a>
    <button
      className="update"
      onClick={[Function]}
    />
    <button
      className="remove"
      onClick={[Function]}
    />
    <input
      autoFocus={true}
      className="list-item-edit"
      data-id="none"
      defaultValue="none"
      onBlur={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter a new name"
    />
  </li>
</div>
`;
